### DISCUSSION TITLE
Introduce contracts for refreshing and revoking tokens

### DISCUSSION THREAD

**Discussion by itambek-godaddy** [SocialiteProviders/Providers#1032] [Ideas]
It looks like a few providers are implementing token refreshing and revoking support, but there is no contract/interface which they implement. This can lead to different APIs across the providers, for example:

* Apple has a `refreshToken` method, which returns the raw `ResponseInterface`
* Mollie has a `getRefreshTokenResponse` method, which returns the parsed body as an array

It would be great to have contracts/interfaces in the Manager repo, so that we would have consistent method names and return values across different providers.

I think these interfaces should be opt-in, so that the individual providers can choose wether to implement them or not.

**Comment by audunru** [comment]
I've been wondering about this as well.

In Socialite's [AbstractProvider](https://github.com/laravel/socialite/blob/f79e67a48c3282aa6331739b0db16f5e1da41d71/src/Two/AbstractProvider.php#L349), there is a `refreshToken` method. It does a standard call to refresh a token, but it just returns the token. As far as I can see, the access token and refresh token which has been set on the user is not updated. So it seems this excercise is left to the reader.
